apps/microtvm-vm/base-box/Vagrantfile.packer-template

```diff
@@ -0,0 +1,45 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+Vagrant.configure("2") do |config|
+  # From hashicorp default template:
+  # https://github.com/hashicorp/packer/blob/master/builder/vagrant/step_create_vagrantfile.go#L23-L37
+
+  config.vm.define "source" do |source|
+    source.vm.box = "{{.SourceBox}}"
+    config.ssh.insert_key = {{.InsertKey}}
```

Comment:
Where do users insert their ssh key and have you thought about security with this ? Is there a requirement to bake that into some image some where ?

Reply:
this is actually not the user's ssh key--it's vagrant/packer's, and it's used to run scripts to provision the VM. this is pretty standard when it comes to scripting VM setup, and further these VMs are intended for local development use only with NAT networking, so I think the risk is minimal. it is required to configure this key, however.

Reply:
Ack. as long as it doesn't get baked into a vcs system I guess that's ok.